crates/hotshot/src/lib.rs

```diff
@@ -292,7 +292,10 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> SystemContext<TYPES, I> {
             locked_view: anchored_leaf.get_view_number(),
             high_qc: initializer.high_qc,
             metrics: Arc::clone(&consensus_metrics),
+            decided_upgrade_cert: None,
```

Comment:
Gating these proved to be impossible. I'll keep a watch for any usage of these vars in coming PRs.

Reply:
Ahh, I see. I had the same question in https://github.com/EspressoSystems/HotShot/pull/3036 and didn't realize it's actually impossible.

Can we instead have two `Consensus` structs with different fields? The code may look ugly but then we don't have to keep a watch to avoid misusage.

Reply:
Is it the upgrade certificates in particular? What were you trying to gate them on?

Reply:
No, it was moving them *into* the Consensus struct. Rust doesn't allow you to gate expressions like this. I actually figured it out after this rolled. Let me tinker around today.